compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -1088,7 +1147,12 @@ static void loadType(MethodVisitor mv, BType bType) {
             loadErrorType(mv, (BErrorType) bType);
             return;
         } else if (bType.tag == TypeTags.UNION) {
-            loadUnionType(mv, (BUnionType) bType);
+            BUnionType unionType = (BUnionType) bType;
+            if (unionType.isTypeDef) {
```

Comment:
This `isTypeDef` is what we'll be moving to a flag right? 

Looping in @pubudu91 for his input too. Basically what we want to do is to have some way to know if this particular union is the type of a type-definition. 

```ballerina
type Cyclic int|Cyclic[];
```

Reply:
You mean the sym tag? That fix was sort of a temporary fix. Basically now every union type symbol will have the `TYPE_DEF` tag on. Don't feel like it's the right solution. That should only be set for unions which are defined like you've illustrated above IMO.

Reply:
No no, what I meant is if we should have a new flag for the type that says this is for a type definition. Instead of a new field in the union type.

Reply:
Yes, but isn't that what the sym tag of the type symbol do (though it seems to be messed up atm)?

Reply:
Shall we use the `Flags.ANONYMOUS` flag for inline unions and not use it for type-definitions, and use that flag to do this differentiation instead of introducing a new field in `BUnionType`?

Reply:
+1, using  Flags.ANONYMOUS